course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/course/CourseManagementActor.java

```diff
@@ -126,9 +129,101 @@ private void createCourse(Request request) throws Exception {
+            String url = getConfigValue(EKSTEP_BASE_URL) + "/content/v3/hierarchy/update";
+            HttpResponse<String> updateResponse =
+                    Unirest.patch(url)
+                            .headers(headers)
+                            .body(mapper.writeValueAsString(generateUpdateHierarchyRequest(request, createResponse)))
+                            .asString();
+            if (null != updateResponse) {
+                Response response = mapper.readValue(updateResponse.getBody(), Response.class);
+                ProjectLogger.log(
+                        "Sized: CourseManagementActor:handleHierarchyData : size of response : "
+                                + updateResponse.getBody().getBytes().length,
+                        LoggerEnum.INFO);
+                if (!StringUtils.equalsIgnoreCase(response.getResponseCode().name(), ResponseCode.OK.name()))
+                    ProjectCommonException.throwClientErrorException(
+                            ResponseCode.customServerError,
+                            MessageFormat.format(
+                                    ResponseCode.customServerError.getErrorMessage(), response.getParams().getErrmsg()));
+            } else {
+                ProjectCommonException.throwClientErrorException(ResponseCode.CLIENT_ERROR);
+            }
+
+        }
+    }
+
+    private Map<String, Object> generateUpdateHierarchyRequest(Request request, Response createResponse) throws Exception {
+        Map<String, Object> nodesModified = new HashMap<String, Object>();
+        Map<String, Object> hierarchy = new HashMap<String, Object>();
+        getRecursiveHierarchyRequest((String) createResponse.getResult().get(SunbirdKey.IDENTIFIER),
+                (List<Map<String, Object>>) request.get(SunbirdKey.HIERARCHY), nodesModified, hierarchy, true);
+        Map<String, Object> updateRequest = new HashMap<String, Object>() {{
+            put(SunbirdKey.REQUEST, new HashMap<String, Object>() {{
+                put(SunbirdKey.DATA, new HashMap<String, Object>() {{
+                    put(SunbirdKey.NODE_MODIFIED, nodesModified);
+                    put(SunbirdKey.HIERARCHY, hierarchy);
+                }});
+            }});
+        }};
+        ProjectLogger.log(
+                "CourseManagementActor:generateUpdateHierarchyRequest : Request for course update Hierarchy : "
+                        + mapper.writeValueAsString(updateRequest),
+                LoggerEnum.INFO.name());
+        return updateRequest;
+    }
+
+    private void getRecursiveHierarchyRequest(String parentId, List<Map<String, Object>> children, Map<String, Object> nodesModified,
+                                              Map<String, Object> hierarchy, Boolean root) {
+        children.forEach(child -> {
+            String identifier = (String) child.get(SunbirdKey.IDENTIFIER);
+            if(!child.containsKey(SunbirdKey.VISIBILITY))
```

Comment:
check child contains list of 4 properties - identifier, mimetype, visibility, contentType